src/Network/HTTP2/Client/Dispatch.hs

```diff
@@ -234,8 +235,15 @@ newHpackEncoderContext encoderBufSize = liftBase $ do
             (\hdrs -> encoder strategy dt buf ptr hdrs)
             (\n -> HPACK.setLimitForEncoding n dt)
   where
+
+#if MIN_VERSION_http2(5,2,0)
+    headerNameCompatibility = CI.original
+#else
+    headerNameCompatibility = id
+#endif
```

Comment:
AFAICT this maintains the old behavior in that we keep case-sensitivity, so as to fix the build.
Probably best to re-evaluate whether the new behavior should be adopted somewhere